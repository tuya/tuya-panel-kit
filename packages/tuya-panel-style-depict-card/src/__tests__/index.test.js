/**
 * @jest-environment jsdom
 */
 import React from 'react';
 import renderer from 'react-test-renderer';
 import TuyaRNSvgs from 'tuya-panel-kit/lib/components/iconfont/svg/defaultSvg';
 import { ClassicDepictCard, NordicDepictCard, AcrylicDepictCard, AcrylicDepictIconCard } from '../index';
 
 describe('DepictCard', () => {
   it('ClassicDepictCard', () => {
     const wrapper = renderer
       .create(<ClassicDepictCard backgroundColor="#fff" />)
       .toJSON();
     expect(wrapper).toMatchSnapshot();
   });
   it('NordicDepictCard', () => {
    const wrapper = renderer
    .create(<NordicDepictCard backgroundColor="#fff" />)
    .toJSON();
  expect(wrapper).toMatchSnapshot();
   })
   it('AcrylicDepictCard', () => {
     const wrapper = renderer.create(<AcrylicDepictCard />).toJSON();
     expect(wrapper).toMatchSnapshot();
   });
   it('AcrylicDepictIconCard', () => {
    const wrapper = renderer.create(<AcrylicDepictIconCard icon={TuyaRNSvgs.power} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
 });
 