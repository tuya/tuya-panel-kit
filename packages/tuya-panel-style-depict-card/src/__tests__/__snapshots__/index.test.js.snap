// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepictCard AcrylicDepictCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 32,
        "flexDirection": "row",
        "paddingBottom": 18,
        "paddingLeft": 40,
        "paddingRight": 50,
        "paddingTop": 30,
        "width": 304,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.5)"
      ellipsizeMode="tail"
      size={24}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 24,
                "fontWeight": "500",
              },
              Array [
                Object {
                  "lineHeight": 48,
                  "marginBottom": 14,
                },
                Object {
                  "lineHeight": 32.6,
                },
              ],
            ],
          ],
        ]
      }
      text="Runtime total"
      type={null}
      weight={500}
    >
      Runtime total
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#505050"
        ellipsizeMode="tail"
        size={48}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#505050",
                  "fontSize": 48,
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 66,
                  },
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={500}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="rgba(0, 0, 0, 0.45)"
        ellipsizeMode="tail"
        size={24}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, 0.45)",
                  "fontSize": 24,
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 32,
                    "paddingBottom": 10,
                  },
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight={400}
      >
        째C
      </Text>
    </View>
  </View>
</View>
`;

exports[`DepictCard AcrylicDepictIconCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 32,
        "flexDirection": "row",
        "paddingBottom": 40,
        "paddingLeft": 40,
        "paddingRight": 40,
        "paddingTop": 38,
        "width": 384,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#158CFB",
          "height": 50,
          "width": 50,
        },
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 50,
          },
          Array [
            Object {
              "marginRight": 24,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 64,
            "width": 64,
          },
          false,
          null,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 64,
              "width": 64,
            },
          ]
        }
        vbHeight={1024}
        vbWidth={1024}
      >
        <RNSVGPath
          d="M512 0C794.760533 0 1024 229.239467 1024 512S794.760533 1024 512 1024 0 794.760533 0 512 229.239467 0 512 0z m185.048178 327.0656a26.988089 26.988089 0 0 0-38.183822 38.183822 207.712711 207.712711 0 1 1-293.728712 0 26.988089 26.988089 0 1 0-38.183822-38.183822c-102.172444 102.1952-102.172444 267.901156 0 370.096356 102.1952 102.172444 267.901156 102.172444 370.096356 0 102.172444-102.1952 102.172444-267.901156 0-370.096356zM511.886222 227.555556a27.079111 27.079111 0 0 0-26.919822 24.302933l-0.136533 2.776178v196.152889a27.079111 27.079111 0 0 0 53.998933 2.776177l0.136533-2.776177v-196.152889a27.079111 27.079111 0 0 0-27.079111-27.079111z"
          fill={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "fill",
            ]
          }
          stroke={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
        />
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.45)"
      ellipsizeMode="tail"
      size={24}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.45)",
                "fontSize": 24,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "lineHeight": 48,
                  "marginBottom": 6,
                },
                Object {
                  "lineHeight": 32,
                },
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={400}
    >
      Current Temp
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="rgba(0, 0, 0, 0.9)"
        ellipsizeMode="tail"
        size={32}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, 0.9)",
                  "fontSize": 32,
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 44,
                  },
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={500}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="rgba(0, 0, 0, 0.5)"
        ellipsizeMode="tail"
        size={24}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 24,
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 32,
                    "paddingBottom": 4,
                  },
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight={400}
      >
        째C
      </Text>
    </View>
  </View>
</View>
`;

exports[`DepictCard ClassicDepictCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "flexDirection": "row",
        "paddingBottom": 36,
        "paddingLeft": 94,
        "paddingRight": 90,
        "paddingTop": 32,
        "width": 516,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(80, 80, 80, 0.5)"
      ellipsizeMode="tail"
      size={24}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(80, 80, 80, 0.5)",
                "fontSize": 24,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "lineHeight": 48,
                  "marginBottom": 12,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={400}
    >
      Current Temp
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#505050"
        ellipsizeMode="tail"
        size={36}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#505050",
                  "fontSize": 36,
                  "fontWeight": "600",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={600}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={36}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 36,
                  "fontWeight": "400",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight={400}
      >
        째C
      </Text>
    </View>
  </View>
</View>
`;

exports[`DepictCard NordicDepictCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 28,
        "flexDirection": "row",
        "paddingBottom": 36,
        "paddingLeft": 94,
        "paddingRight": 90,
        "paddingTop": 32,
        "width": 516,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.5)"
      ellipsizeMode="tail"
      size={24}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 24,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "lineHeight": 48,
                  "marginBottom": 12,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={400}
    >
      Current Temp
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={40}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 40,
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 64,
                  },
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={500}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={36}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 36,
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                  },
                  Object {
                    "lineHeight": 64,
                  },
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight={500}
      >
        째C
      </Text>
    </View>
  </View>
</View>
`;
